src/polygon.jl

```diff
@@ -0,0 +1,69 @@
+"""
+    getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int) 
+Get layer of the desired level from the dataset
+"""
+function getlevel(dataset::ArchGDAL.AbstractDataset, reqlevel::Int)
+    # number of layers in the dataset
+    layers = ArchGDAL.nlayer(dataset)
+
+    for i = 0:layers-1
+        layer = ArchGDAL.getlayer(dataset, i)
+        layername = ArchGDAL.getname(layer)
+        layerlevel = split(layername, "_")[end]
+        if layerlevel == string(reqlevel)
+            return layer
+        end
+    end
```

Comment:
Why loop if you are only returning layer i?

Reply:
There are multiple layers with names like `gadm_ind36_0`, `gadm_ind36_1`, and they are not necessarily ordered. Hence, I need to loop over all layers to see which one is required. Also, it is not necessary that `gadm_ind36` will be the first element. Therefore looping seems to be the simplest solution.

Reply:
Yes, let's get it started. I think we will need to brainstorm a more user friendly API after we get this one merged :+1: